scipy/stats/_multivariate.py

```diff
@@ -3275,12 +3275,13 @@ def __call__(self, n, p, seed=None):
         """
         return multinomial_frozen(n, p, seed)
 
-    def _process_parameters(self, n, p, eps=1e-15):
+    def _process_parameters(self, n, p):
         """Returns: n_, p_, npcond.
 
         n_ and p_ are arrays of the correct shape; npcond is a boolean array
         flagging values out of the domain.
         """
+        eps = np.finfo(np.result_type(np.asarray(p), np.float32)).eps * 10
```

Comment:
```suggestion
        eps = np.finfo(np.result_type(np.asarray(p), np.float16)).eps * 10
```
maybe?

Reply:
can this be just
`eps = xp.finfo(xp_default_dtype(xp)).eps * 10   # or other wiggle factor`

This will not work currently since `xp_default_dtype` only account for torch defaulting to f32; 
Assuming that the failure mode is that jax returns f32 array even when explicitly asked for an f64,  `xp_default_dtype` should grow some logic to account for JAX idiosyncrasy.

Reply:
There are tests that are passing integers like `[0, 1]` or something.

Reply:
Ok then, that figures. Then indeed, there's no way around `result_type`. Sorry for the noise